designs/0005-static-matrices.md

```diff
@@ -0,0 +1,240 @@
+- Feature Name: static_matrices
+- Start Date: April 30th, 2020
+- RFC PR: [Example Branch](https://github.com/stan-dev/math/compare/feature/var-template)
+- Stan Issue: [#1805](https://github.com/stan-dev/math/issues/1805)
+
+# Summary
+[summary]: #summary
+
+This proposes a generalization of the reverse mode autodiff type in Stan Math,
+`var_value<T>`, where `T` is the underlying storage type of the autodiff variable.
+For `T == double`, this type is equivalent to the current Stan Math `var` type.
+
+The usefulness of this type comes from setting `T` to more general data structures,
+in particular the Eigen Matrix types. Instead of dealing with matrices of vars,
+this allows operations in the Math library to be written in terms of vars that are
+themselves matrices.
+
+This makes working with matrix variables in Stan autodiff much faster. This design
+is fully backwards compatible with the curren Stan autodiff types and those remain.
+
+For the rest of this proposal, "autodiff" means "reverse mode autodiff".
```

Comment:
Two things:
1. Typo: the period goes within the quotation mark... so the correction is: `"reverse mode autodiff."`
2. It'd be good to add that the rest of the proposal, we'll leave off the `stan::math` namespace. Suggested sentence: "For brevity, we will omit the `stan::math` namespace where it's clear from the context." (or something like that?)